### PR TITLE
hw05

### PR DESCRIPTION
**Pull request by u2107680007911543774e7r**
another attempt

### CODE REVIEW

**Review by Yanetta — APPROVED** [review]
good

File: src/main/java/homework05/Main.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	public static void main(String[] args) {
+		Fruit apple1 = new Fruit(Type.APPLE, 30, LocalDate.of(2019, Month.JANUARY, 1), 40);
+		Fruit apple2 = new Fruit(Type.APPLE, 60, LocalDate.of(2018, Month.DECEMBER, 11), 55);
+		Fruit banana1 = new Fruit(Type.BANANA, 30, LocalDate.of(2018, Month.NOVEMBER, 29), 30);
+		Fruit strawberry1 = new Fruit(Type.STRAWBERRY, 30, LocalDate.of(2019, Month.JANUARY, 3), 300);
+		Fruit pear1 = new Fruit(Type.PEAR, 30, LocalDate.of(2004, Month.APRIL, 4), 40);
+		Fruit pear2 = new Fruit(Type.PEAR, 60, LocalDate.of(2018, Month.DECEMBER, 4), 60);
+		
+		List<Fruit> list = Arrays.asList(apple1, apple2, banana1, strawberry1, pear1, pear2);
+		Shop shop = new Shop("SILPO", list);
```
Comment (SerhiiPasko) [inline src/main/java/homework05/Main.java:19]: =)

File: src/main/java/homework05/Main.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		
+		System.out.println(shop.allFruitOfType(Type.PEAR));
+		System.out.println(shop.allFreshAndType(Type.APPLE, LocalDate.now()));
+		System.out.println(shop.allFresh(LocalDate.now()));
+		System.out.println(shop.checkExpireDate(45));
+		shop.checkExpireDate(45).stream().map(f->f.getType().getCountry()).forEach(System.out::println);
```
Comment (SerhiiPasko) [inline src/main/java/homework05/Main.java]: Why you write some logic there?

File: src/main/java/homework05/Shop.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+	public List<Fruit> checkExpireDate(int checkDays) {
+		return fruits.stream().filter(f -> f.getExpireDays() > checkDays).collect(Collectors.toList());
+	}
+
+	public List<Fruit> allFreshAndType(Type type, LocalDate expDate) {
+		return allFruitOfType(type).stream().filter(f -> f.isExpired(expDate)).collect(Collectors.toList());
+	}
+
+	public List<Fruit> reSetPricexRealization(Type ... types) {
```
Comment (SerhiiPasko) [inline src/main/java/homework05/Shop.java]: You use magic value "10". Read homeTask. It is must be a param. Also in this method you change your internal List<Fruits>. What will be happen when you use this method twice? When you use stream...collect you have new list. There return new list too

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Good (80%). Read comments

File: src/main/java/homework05/Shop.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	public List<Fruit> checkExpireDate(int checkDays) {
+		return fruits.stream().filter(f -> f.getExpireDays() > checkDays).collect(Collectors.toList());
+	}
+
+	public List<Fruit> allFreshAndType(Type type, LocalDate expDate) {
+		return allFruitOfType(type).stream().filter(f -> f.isExpired(expDate)).collect(Collectors.toList());
+	}
+
+	public List<Fruit> reSetPricexRealization(int percent, Type ... types) {
+		List<Fruit> newList = fruits;
```
Comment (SerhiiPasko) [inline src/main/java/homework05/Shop.java]: You don't create new List. You have new link to your list. In this case you change your input list

Comment (SerhiiPasko) [reply src/main/java/homework05/Shop.java]: Must be List\<Fruit\> newList=new ArrayList\<\>()

File: src/main/java/homework05/Type.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package homework05;
+
+public enum Type {
+	APPLE("Italy"), STRAWBERRY("Poland"), PEAR("Ukraine"), ORANGE("Spain"), PINEAPPLE("Turkey"), BANANA("Ecuador");
+
+	String country;
```
Comment (SerhiiPasko) [inline src/main/java/homework05/Type.java]: private

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Read comments. You have critical mistake

File: src/main/java/homework05/Type.java (hunk 5 above)
Comment (u2107680007911543774e7r) [reply src/main/java/homework05/Type.java]: I use country in toString of Fruit class...

File: src/main/java/homework05/Shop.java (hunk 4 above)
Comment (u2107680007911543774e7r) [reply src/main/java/homework05/Shop.java]: Can I do next:
`List<Fruit> newList = new ArrayList<Fruit>(fruits);`

Comment (SerhiiPasko) [reply src/main/java/homework05/Shop.java]: Yes

**Pull request merged**